File: app/src/main/java/in/completecourse/questionbank/MoreAppsActivity.java
package in.completecourse.questionbank;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.ProgressBar;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.google.firebase.firestore.FirebaseFirestore;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import in.completecourse.questionbank.adapter.MoreAppsAdapter;
import in.completecourse.questionbank.helper.HelperMethods;
import in.completecourse.questionbank.model.MoreAppsItem;

public class MoreAppsActivity extends AppCompatActivity {
    @BindView(R.id.recycler_view)
    RecyclerView recyclerView;
    @BindView(R.id.progressBar)
    ProgressBar progressBar;
    private ArrayList<MoreAppsItem> moreAppsItems;
    private FirebaseFirestore db;
    MoreAppsAdapter adapter;
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_more_apps);

        ButterKnife.bind(this);

        db = FirebaseFirestore.getInstance();

        moreAppsItems = new ArrayList<>();

        adapter = new MoreAppsAdapter(this, moreAppsItems);
        recyclerView.setLayoutManager(new LinearLayoutManager(this, RecyclerView.VERTICAL, false));
        recyclerView.addItemDecoration(new DividerItemDecoration(this, RecyclerView.VERTICAL));
        recyclerView.setAdapter(adapter);

        getMoreApps();

        recyclerView.addOnItemTouchListener(new HelperMethods.RecyclerTouchListener(this, new HelperMethods.ClickListener() {
            @Override
            public void onClick(int position) {
                Intent intent = new Intent("android.intent.action.VIEW", Uri.parse(moreAppsItems.get(position).getApp_play_url()));
                startActivity(intent);
            }
        }));
    }

    private void getMoreApps(){
        db.collection("more_apps_qb").get()
                .addOnCompleteListener(task -> {
                    progressBar.setVisibility(View.GONE);
                }).addOnSuccessListener(queryDocumentSnapshots -> {
            List<MoreAppsItem> items = queryDocumentSnapshots.toObjects(MoreAppsItem.class);
            moreAppsItems.addAll(items);
            adapter.setItems(moreAppsItems);
            adapter.notifyDataSetChanged();
        }).addOnFailureListener(e -> Toast.makeText(MoreAppsActivity.this, e.getMessage(), Toast.LENGTH_LONG).show());
    }

}

File: app/src/main/java/in/completecourse/questionbank/MoreAppsActivity.kt
package `in`.completecourse.questionbank

import `in`.completecourse.questionbank.adapter.MoreAppsAdapter
import `in`.completecourse.questionbank.helper.HelperMethods
import `in`.completecourse.questionbank.model.MoreAppsItem
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.View
import android.widget.ProgressBar
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import butterknife.BindView
import butterknife.ButterKnife
import com.google.android.gms.tasks.Task
import com.google.firebase.firestore.FirebaseFirestore
import com.google.firebase.firestore.QuerySnapshot
import kotlinx.android.synthetic.main.activity_more_apps.*
import java.util.*

class MoreAppsActivity : AppCompatActivity() {

    private var moreAppsItems: ArrayList<MoreAppsItem>? = null
    var adapter: MoreAppsAdapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_more_apps)

        moreAppsItems = ArrayList<MoreAppsItem>()
        adapter = MoreAppsAdapter(this, moreAppsItems!!)
        recycler_view.layoutManager = LinearLayoutManager(this, RecyclerView.VERTICAL, false)
        recycler_view.addItemDecoration(DividerItemDecoration(this, RecyclerView.VERTICAL))
        recycler_view.adapter = adapter

        moreApps

        recycler_view.addOnItemTouchListener(HelperMethods.RecyclerTouchListener(this, object : HelperMethods.ClickListener {
            override fun onClick(position: Int) {
                val intent = Intent("android.intent.action.VIEW", Uri.parse(moreAppsItems!![position].app_play_url))
                startActivity(intent)
            }
        }))
    }

    private val moreApps: Unit get() {
            FirebaseFirestore.getInstance().collection("more_apps_qb").get()
                    .addOnCompleteListener {
                        progressBar?.visibility = View.GONE }
                    .addOnSuccessListener { queryDocumentSnapshots: QuerySnapshot ->
                        val items: List<MoreAppsItem> = queryDocumentSnapshots.toObjects<MoreAppsItem>(MoreAppsItem::class.java)
                        moreAppsItems?.addAll(items)
                        adapter?.setItems(moreAppsItems!!)
                        adapter?.notifyDataSetChanged()
                    }.addOnFailureListener { e: Exception ->
                        Toast.makeText(this@MoreAppsActivity, e.message, Toast.LENGTH_LONG).show() }
        }
}